cli/dcoscli/main.py

```diff
@@ -67,17 +64,31 @@ def _main():
             msg = 'Unable to get the cluster_id of the cluster.'
             logger.exception(msg)
 
+    # send args call to segment.io
+    reporting_executor = ThreadPoolExecutor(max_workers=2)
+    report = config.get('core.reporting', True)
+    if report and dcoscli.version != 'SNAPSHOT':
+        analytics.segment_track_cli(reporting_executor, config, cluster_id)
+
     # the call to retrieve cluster_id must happen before we run the subcommand
     # so that if you have auth enabled we don't ask for user/pass multiple
     # times (with the text being out of order) before we can cache the auth
     # token
-    subproc = Popen([executable,  command] + args['<args>'],
-                    stderr=PIPE)
-
-    if dcoscli.version != 'SNAPSHOT':
-        return analytics.wait_and_track(subproc, cluster_id)
+    if command in subcommand.default_subcommands():
+        sys.argv[1:] = [command] + args['<args>']
```

Comment:
I see. You are doing this since we need to remove `[options]` from the args that we pass to `docopt`. Instead of changing `sys.argv` why don't we just pass the array of string explicitly to `docopt`? They have an `argv` parameter that you can override.

Also, let's move this code to `subcommand.SubcommandMain`.